### ISSUE TITLE
Unable to uninstall Microsoft Visual C++ packages

### ISSUE THREAD

**Issue by ppvnf** [martinet101/WingetUI#203]
It seems Microsoft is changing the names of Microsoft Visual C++ packages, which may be the reason why it's not possible to uninstall them through WingetUI (as admin and interactive also don't work). Installing works fine, though.

This is likely a WIP, since there are still duplicates of Microsoft Visual C++ 2008 packages in CLI, but I was able to successfully uninstall all Microsoft Visual C++ packages through CLI, while WingetUI failed for all of them. 

Besides, WingetUI doesn't correctly identify the package source as Winget and the package ID of some of them is messed up on WingetUI (but not on winget-cli):

![Captura de tela 2022-10-05 132855](https://user-images.githubusercontent.com/73800734/194112851-c694c4e9-da57-44c1-9a27-36ca43a26e19.png)
![Captura de tela 2022-10-05 133913](https://user-images.githubusercontent.com/73800734/194114828-a4434cb2-966b-47f9-946e-80a9b73828da.png)
![Captura de tela 2022-10-05 134235](https://user-images.githubusercontent.com/73800734/194115483-70294921-be21-493a-af9d-ca24ace0e0bd.png)
![Captura de tela 2022-10-05 133041](https://user-images.githubusercontent.com/73800734/194113170-1f6741a7-afc9-4d77-ab18-74d2976b5671.png)
![Captura de tela 2022-10-05 135603](https://user-images.githubusercontent.com/73800734/194117990-acb406c0-18c5-48f1-aca9-4f664332a755.png)
![Captura de tela 2022-10-05 134042](https://user-images.githubusercontent.com/73800734/194119363-7a0f0868-8040-4894-9d36-43674a203402.png)

Possibly related to https://github.com/martinet101/WingetUI/issues/198

**Comment by ppvnf** [comment]
Same problem with Microsoft.DotNet.DesktopRuntime:
I am unable to uninstall Microsoft.DotNet.DesktopRuntime.3_1 and Microsoft.DotNet.DesktopRuntime.6 using WingetUI, but through Winget-cli it is possible.
For Microsoft.DotNet.DesktopRuntime.5, it's impossible to uninstall using both Winget and Winget-cli, since it's not recognized by Winget-cli...
It's kinda ironic Microsoft products having problems with Winget. Anyway, I am gonna complain to Microsoft. I am not sure there's much you can do though

**Comment by martinet101** [comment]
Yes, this is winget-cli things. .NET and C++Redist have always caused problems.

**Comment by martinet101** [comment]
I have managed to get the some ids to fully show in the uninstall section, could you please try uninstalling again with the latest head?

**Comment by ppvnf** [comment]
> Yes, this is winget-cli things. .NET and C++Redist have always caused problems.

I think this is the same bug as https://github.com/martinet101/WingetUI/issues/198:

`winget uninstall --id Microsoft.VCRedist.2005.x64 -e --silent` 

fails, but

`winget uninstall --id Microsoft.VCRedist.2005.x64 -e` 

does not, so I am gonna suggest again to automatically try to uninstall without silent parameter as a fallback.

I've also noticed that WingetUI install message doesn't show the full package name in this case, couldn't this also be a problem?

![Captura de tela 2022-10-06 072251](https://user-images.githubusercontent.com/73800734/194289948-9ae72a08-dee6-4852-82b6-c971e9d7f365.png)


> I have managed to get the some ids to fully show in the uninstall section, could you please try uninstalling again with the latest head?

Anyhow, I've tried to reinstall 2005/2008 x86 and x64 versions and I don't get it why they are showing fine on Winget-cli, but messed up on WingetUI (after install they appear fine initially, however refreshing makes them like this again):

![Captura de tela 2022-10-06 071620](https://user-images.githubusercontent.com/73800734/194289307-c786caa0-92d8-4c51-b452-d10006d3c3bf.png)
![Captura de tela 2022-10-06 072322](https://user-images.githubusercontent.com/73800734/194289665-e05adeb7-4ece-4234-9168-3b61e5fe57f0.png)

I've opened issues on Winget Github page about .NET and C++Redist, so hopefully someone from Microsoft will fix some of those problems on their end.
